File: FragmentCommunicationWE/app/src/main/java/com/example/fragmentcommunicationwe/StudentDetailsFragment.java
package com.example.fragmentcommunicationwe;

import android.os.Bundle;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.Button;
import android.widget.EditText;

import androidx.annotation.NonNull;
import androidx.annotation.Nullable;
import androidx.fragment.app.Fragment;

public class StudentDetailsFragment extends Fragment {

    private EditText etName, etAge;
    private Button btnNext;
    private Listener listener;


    @Override
    public View onCreateView(LayoutInflater inflater, ViewGroup container,
                             Bundle savedInstanceState) {
        // Inflate the layout for this fragment
        return inflater.inflate(R.layout.fragment_student_details, container, false);

    }


    @Override
    public void onViewCreated(@NonNull View view, @Nullable Bundle savedInstanceState) {
        super.onViewCreated(view, savedInstanceState);
        initViews(view);

    }

    private void initViews(View view) {
        etName = view.findViewById(R.id.etName);
        etAge = view.findViewById(R.id.etAge);
        btnNext = view.findViewById(R.id.btnNext);

        btnNext.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View v) {
                String name = etName.getText().toString();
                int age = Integer.parseInt(etAge.getText().toString());
                Bundle bundle = new Bundle();
                bundle.putString("name", name);
                bundle.putInt("age", age);
                if (listener != null){
                    listener.launchPerformanceFragment(bundle);
                }
            }
        });
    }

    public void setListener(Listener listener){
        this.listener = listener;
    }
}

File: Kotlin Projects/NavigationGraphIW/app/src/main/java/com/example/navigationgraph_iw/Fragments/StudentDetailsFragment.kt
package com.example.navigationgraph_iw.Fragments

import android.os.Bundle
import android.view.View
import androidx.fragment.app.Fragment
import androidx.navigation.NavController
import androidx.navigation.Navigation
import com.example.navigationgraph_iw.R
import kotlinx.android.synthetic.main.fragment_student_details.*

class StudentDetailsFragment : Fragment(R.layout.fragment_student_details) {

    private lateinit var navController: NavController

    override fun onViewCreated(view: View, savedInstanceState: Bundle?) {
        super.onViewCreated(view, savedInstanceState)

        navController = Navigation.findNavController(view)

        btnNext.setOnClickListener {
            val name = etName.text.toString()
            val age = etAge.text.toString().toInt()
            val bundle = Bundle()
            bundle.putString("name", name)
            bundle.putInt("age", age)
            navController.navigate(R.id.action_studentDetailsFragment_to_studentPerformanceFragment, bundle)
        }
    }
}